launcher/src/main/java/org/wildfly/core/launcher/StandaloneCommandBuilder.java

```diff
@@ -153,8 +163,18 @@ public StandaloneCommandBuilder addJavaOptions(final String... javaOpts) {
      */
     public StandaloneCommandBuilder addJavaOptions(final Iterable<String> javaOpts) {
         if (javaOpts != null) {
-            for (String javaOpt : javaOpts) {
-                addJavaOption(javaOpt);
+            final Iterator<String> i = javaOpts.iterator();
+            for (String javaOpt = i.next(); i.hasNext(); javaOpt = i.next()) {
+                if (Arguments.isModuleSystemArgument(javaOpt)) {
+                    // handle java module system options
+                    if (i.hasNext()) {
+                        this.javaOpts.add(Arguments.JavaModuleSystemArgument.of(javaOpt, i.next()));
+                    } else {
+                        // ignore corrupted module system argument
```

Comment:
Same as above, we should just add the arguments and allow the command to fail when executed.